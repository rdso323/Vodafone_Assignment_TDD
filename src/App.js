import "./App.css";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Albums from "./components/Albums";
import Photos from "./components/Photos";
import React, { useState, useEffect } from "react";

function App() {
  const [data_albums, setAlbums] = useState([null]);
  const [data_users, setUsers] = useState(null);

  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/albums")
      .then((res) => res.json())
      .then((data_albums) => setAlbums(data_albums));

    fetch("https://jsonplaceholder.typicode.com/users")
      .then((res) => res.json())
      .then((data_users) => setUsers(data_users));
  });

  return (
    <Router>
      <Switch>
        <div className="App">
          <Route exact path="/">
            <Albums data_albums={data_albums} data_users={data_users} />
          </Route>
          <Route path="/pics/:topic" component={Photos}></Route>
        </div>
      </Switch>
    </Router>
  );
}

export default App;
