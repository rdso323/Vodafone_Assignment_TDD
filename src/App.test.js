import React from "react";
import { mount } from "enzyme";
import App from "./App";

describe("App component snapshot testing", () => {
  // Snapshot Testing for App component
  test("Matches the snapshot", () => {
    const component = mount(<App />);
    expect(component.getElements()).toMatchSnapshot();
  });

  test("App component is displayed", () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(App)).toHaveLength(1);
  });
});
