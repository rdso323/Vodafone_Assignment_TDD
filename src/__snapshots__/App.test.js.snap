// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot testing Matches the snapshot 1`] = `
Array [
  <App />,
]
`;
